src/Nethermind/Nethermind.Config/ConfigSourceHelper.cs

```diff
@@ -86,13 +103,76 @@ public static object GetDefault(Type type)
             return type.IsValueType ? (false, Activator.CreateInstance(type)) : (false, null);
         }
 
+        public static bool TryFromHex(Type type, string itemValue, out object value)
+        {
+            if (!itemValue.StartsWith("0x"))
+            {
+                value = null;
+                return false;
+            }
+            switch (Type.GetTypeCode(type))
+            {
+                case TypeCode.Byte:
+                    value = Convert.ToByte(itemValue, 16);
+                    return true;
+                case TypeCode.SByte:
+                    value = Convert.ToSByte(itemValue, 16);
+                    return true;
+                case TypeCode.UInt16:
+                    value = Convert.ToUInt16(itemValue, 16);
+                    return true;
+                case TypeCode.UInt32:
+                    value = Convert.ToUInt32(itemValue, 16);
+                    return true;
+                case TypeCode.UInt64:
+                    value = Convert.ToUInt64(itemValue, 16);
+                    return true;
+                case TypeCode.Int16:
+                    value = Convert.ToInt16(itemValue, 16);
+                    return true;
+                case TypeCode.Int32:
+                    value = Convert.ToInt32(itemValue, 16);
+                    return true;
+                case TypeCode.Int64:
+                    value = Convert.ToInt64(itemValue, 16);
+                    return true;
+                default:
+                    value = null;
+                    return false;
+            }
+        }
+
         private static object GetValue(Type valueType, string itemValue)
         {
+            if (Nullable.GetUnderlyingType(valueType) is var nullableType && nullableType is not null)
+            {
+                if (string.IsNullOrEmpty(itemValue) || itemValue.Equals("null", StringComparison.InvariantCultureIgnoreCase))
+                {
+                    return null;
+                }
+
+                return GetValue(nullableType, itemValue);
+            }
```

Comment:
This is pretty much the same check as in `ParseValue`. Any chance to refactor to a common utility?